### PR TITLE
Massive api changes

### PR DESCRIPTION
**Pull request by andriy332**

### CODE REVIEW

**Comment by andriycraft** [comment]
v

File: app.js [hunk 1]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
+      info('Returning default robots.txt page (because disallow crawlers is on)')
+      statusCode(200)
+      sendHeader('Content-type', 'text/plain')
       writeContent('User-agent: *')
       endResponse('Disallow: /')
     } else {
-      readfile()
+      readFile()
     }
   } else if (req.url === '/login.html/') {
```
Comment (andriycraft) [inline app.js:169]: Remove.

File: configs/config.json [hunk 2]
```diff
@@ -2,14 +2,13 @@
...
-	"authentication_password" : "XiLi1A@IXK381c!#@&kz,x,1.12Z3A12X2AZ1xX1",
+	"authentication_username" : "admin",
+	"authentication_password" : "admin",
 	"authentication_realm" : "KinashServer",
 	"authentication_file" : "./public_html/login.html",
 	"max_url_length": "500",
 	"max_url_length_tip_do_not_change": "Pro Tip: Use only numbers in max_url_length to prevent server crash",
 	"error500page": "<html><head><title>Error: 500</title><link href='https://fonts.googleapis.com/css2?family=Roboto&amp;display=swap' rel='stylesheet'></head><body><center style='margin-top: 15%'><h1 style='color: #401542; font-family: Roboto'>500</h1><br><br><p>Oops! An unknown error happend</p></center></body></html>",
-	"errornoindexpage": "Error: No index file found! Please create a new one",
 	"error500noauthpage": "Error: No authentication file found",
```
Comment (andriycraft) [inline configs/config.json:12]: Remove

File: configs/default-config.json [hunk 3]
```diff
@@ -2,14 +2,13 @@
...
-	"authentication_password" : "XiLi1A@IXK381c!#@&kz,x,1.12Z3A12X2AZ1xX1",
+	"authentication_username" : "admin",
+	"authentication_password" : "admin",
 	"authentication_realm" : "KinashServer",
 	"authentication_file" : "./public_html/login.html",
 	"max_url_length": "500",
 	"max_url_length_tip_do_not_change": "Pro Tip: Use only numbers in max_url_length to prevent server crash",
 	"error500page": "<html><head><title>Error: 500</title><link href='https://fonts.googleapis.com/css2?family=Roboto&amp;display=swap' rel='stylesheet'></head><body><center style='margin-top: 15%'><h1 style='color: #401542; font-family: Roboto'>500</h1><br><br><p>Oops! An unknown error happend</p></center></body></html>",
-	"errornoindexpage": "Error: No index file found! Please create a new one",
 	"error500noauthpage": "Error: No authentication file found",
```
Comment (andriycraft) [inline configs/default-config.json:12]: Remove too

File: app.js [hunk 4]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
-    if (err === '401') { res.end(config.error401page); return }
-    if (err === '403') { res.end(config.error403page); return }
-    if (err === '404') { res.end(config.error404page); return }
-    if (err === '414') { res.end(config.error414page); return }
-    if (err === '500') { res.end(config.error500page); return }
-    writeContent('<h3>Error</h3>')
-    writeContent(`<p>Error code: ${err}</p>`)
-    endResponse(`<u>${message}</u>`)
+    if (err === 400 || message === "null") {
+		res.end(config.error400page)
```
Comment (andriycraft) [inline app.js:47]: endResponse(config.error400page)

File: app.js [hunk 5]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
+		return 
+	}
+    if (err === 401 || message === "null") {
+		res.end(config.error401page)
```
Comment (andriycraft) [inline app.js:51]: endResponse(config.error401page)

File: app.js [hunk 6]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
+		return 
+	}
+    if (err === 403 || message === "null") {
+		res.end(config.error403page)
```
Comment (andriycraft) [inline app.js:55]: endResponse(config.error403page)

File: app.js [hunk 7]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
+		return 
+	}
+    if (err === 404 || message === "null") {
+		res.end(config.error404page)
```
Comment (andriycraft) [inline app.js:59]: endResponse(config.error404page)

File: app.js [hunk 8]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
+		return 
+	}
+    if (err === 405 || message === "null") {
+		res.end('Unsupported method')
```
Comment (andriycraft) [inline app.js:63]: endResponse('Unsupported method')

File: app.js [hunk 9]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
+		return 
+	}
+    if (err === 414 || message === "null") {
+		res.end(config.error414page)
```
Comment (andriycraft) [inline app.js:67]: endResponse(config.error414page)

File: app.js [hunk 10]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
+		return 
+	}
+    if (err === 500 || message === "null") {
+		res.end(config.error500page)
```
Comment (andriycraft) [inline app.js:71]: endResponse(config.error500page)

File: app.js [hunk 11]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
+		return 
+	}
+	else{
+		writeContent('<center>')
+		writeContent(`<h1>${err} ${statusText}</h1>`)
+		writeContent(`<p>${message}</p>`)
+		writeContent('</center>')
+		endResponse()
```
Comment (andriycraft) [inline app.js:79]: Remove

File: app.js [hunk 12]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
+	}
+    if (err === 500 || message === "null") {
+		res.end(config.error500page)
+		return 
+	}
+	else{
+		writeContent('<center>')
+		writeContent(`<h1>${err} ${statusText}</h1>`)
+		writeContent(`<p>${message}</p>`)
+		writeContent('</center>')
```
Comment (andriycraft) [inline app.js:78]: `endResponse('</center>')`

File: app.js [hunk 13]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
-    console.warn('\x1b[33m [WARN] Please report this bug to our github')
-    console.warn('\x1b[33m [WARN] ERROR:')
-    console.warn('\x1b[33m' + err + '\x1b[0m')
-    errorlog.error('[ERROR] An error handling in user request')
-    errorlog.warn('[WARN] Please report this bug to our github')
-    errorlog.warn('[WARN] ERROR:')
-    errorlog.warn(err)
+    error('Error handling this user request!')
+    error('Please report it to our github: https://github.com/andriy332/KinashServer/')
+    error('ERROR: ' + err)
```
Comment (andriycraft) [inline app.js:105]: error(err)

File: app.js [hunk 14]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
+    warning('Do not forget to share full server log')
   })
-  log.log('[INFO] ' + '[' + req.socket.remoteAddress + '] ' + Date() + ' ' + req.method + ' ' + req.url)
-  console.log('[INFO] ' + '\x1b[0m\x1b[32m [' + req.socket.remoteAddress + '] ' + Date() + ' ' + req.method + ' ' + req.url)
-  if (req.url === '/') {
+  
+  info('[' + req.socket.remoteAddress + '] ' + Date() + ' ' + req.method + ' ' + req.url)
+  
+  if(req.method ===! 'GET'){
+    returnError(405, null)
```
Comment (andriycraft) [inline app.js:112]: returnError(405, null, null)

File: app.js [hunk 15]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
+            sendHeader('Content-type', 'text/html')
             endResponse(data)
           } else {
-            returnError(404, null)
+            returnError(404, null, null)
           }
         })
       } else {
-        returnError(503, 'No index page')
+        returnError(500, 'Default index.html file is missing')
```
Comment (andriycraft) [inline app.js:128]: returnError(500, 'index.html file is missing', null)

File: app.js [hunk 16]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
     const b64auth = (req.headers.authorization || '').split(' ')[1] || ''
     const [login, password] = Buffer.from(b64auth, 'base64').toString().split(':')
     if (login && password && login === auth.login && password === auth.password) {
       fs.readFile(config.authentication_file, 'utf8', function (err, data) {
         if (err) {
-          returnError(503, 'No authentication file found.')
-          console.warn('\x1b[33m[WARN] User ' + req.socket.remoteAddress + ' passed the authentication with but the authentication file doesnt exists \x1b[0m\x1b[32m')
+          returnError(500, 'Authentication file is missing.')
+          warning('User ' + req.socket.remoteAddress + ' passed the authentication')
+          error('Error: the authentication file is missing')
```
Comment (andriycraft) [inline app.js:142]: Error: The authentication file is missing

File: app.js [hunk 17]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
+        warning('User ' + req.socket.remoteAddress + ' passed the authentication')
       })
       return
     }
     res.setHeader('WWW-Authenticate', 'Basic realm="' + config.authentication_realm + '"')
     returnError(401, null)
-    console.warn('\x1b[33m[WARN] User ' + req.socket.remoteAddress + ' is tried to login (or failed the authentication)')
+    warning('User ' + req.socket.remoteAddress + ' is tried to login (or failed the authentication)')
   } else if (req.url.length > config.max_url_length) {
     returnError(414, null)
```
Comment (andriycraft) [inline app.js:154]: returnError(414, null, null)

File: app.js [hunk 18]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
   } else if (req.url === '/login.html') {
-    returnError(404, null)
+    returnError(403, null)
+    warning('Denied user request ' + req.socket.remoteAddress + ' to ' + req.url)
+    warning('Reason: This page is protected')
   } else if (req.url === '/robots.txt') {
     if (config.disallowcrawlers === 'true') {
-      res.writeHead(200, { 'Content-Type': 'text/plain' })
+      info('Returning default robots.txt page (because disallow crawlers is on)')
+      statusCode(200)
```
Comment (andriycraft) [inline app.js:162]: Remove

File: app.js [hunk 19]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
     }
     res.setHeader('WWW-Authenticate', 'Basic realm="' + config.authentication_realm + '"')
     returnError(401, null)
-    console.warn('\x1b[33m[WARN] User ' + req.socket.remoteAddress + ' is tried to login (or failed the authentication)')
+    warning('User ' + req.socket.remoteAddress + ' is tried to login (or failed the authentication)')
   } else if (req.url.length > config.max_url_length) {
     returnError(414, null)
   } else if (req.url === '/login.html') {
-    returnError(404, null)
+    returnError(403, null)
```
Comment (andriycraft) [inline app.js:156]: returnError(403, null, null)

File: app.js [hunk 20]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
     if (login && password && login === auth.login && password === auth.password) {
       fs.readFile(config.authentication_file, 'utf8', function (err, data) {
         if (err) {
-          returnError(503, 'No authentication file found.')
-          console.warn('\x1b[33m[WARN] User ' + req.socket.remoteAddress + ' passed the authentication with but the authentication file doesnt exists \x1b[0m\x1b[32m')
+          returnError(500, 'Authentication file is missing.')
+          warning('User ' + req.socket.remoteAddress + ' passed the authentication')
+          error('Error: the authentication file is missing')
         }
         res.writeHead(200, { 'Content-Type': 'text/html' })
```
Comment (andriycraft) [inline app.js:144]: sendHeader(''Content-type, 'text/html')

File: app.js [hunk 21]
```diff
@@ -10,107 +10,172 @@ if (!fs.existsSync(folder)) {
...
   } else if (req.url === config.authentication_url) {
     const auth = { login: config.authentication_username, password: config.authentication_password }
     const b64auth = (req.headers.authorization || '').split(' ')[1] || ''
     const [login, password] = Buffer.from(b64auth, 'base64').toString().split(':')
     if (login && password && login === auth.login && password === auth.password) {
       fs.readFile(config.authentication_file, 'utf8', function (err, data) {
         if (err) {
-          returnError(503, 'No authentication file found.')
-          console.warn('\x1b[33m[WARN] User ' + req.socket.remoteAddress + ' passed the authentication with but the authentication file doesnt exists \x1b[0m\x1b[32m')
+          returnError(500, 'Authentication file is missing.')
```
Comment (andriycraft) [inline app.js:140]: ` returnError(500, null, null)`

File: configs/default-config.json [hunk 22]
```diff
@@ -2,14 +2,13 @@
...
 	"port" : "80",
 	"authentication_url" : "/authentication",
-	"authentication_username" : "KinashServer",
-	"authentication_password" : "XiLi1A@IXK381c!#@&kz,x,1.12Z3A12X2AZ1xX1",
+	"authentication_username" : "admin",
+	"authentication_password" : "admin",
 	"authentication_realm" : "KinashServer",
 	"authentication_file" : "./public_html/login.html",
 	"max_url_length": "500",
 	"max_url_length_tip_do_not_change": "Pro Tip: Use only numbers in max_url_length to prevent server crash",
```
Comment (andriycraft) [inline configs/default-config.json:10]: Remove

File: configs/config.json [hunk 23]
```diff
@@ -2,14 +2,13 @@
...
 	"port" : "80",
 	"authentication_url" : "/authentication",
-	"authentication_username" : "KinashServer",
-	"authentication_password" : "XiLi1A@IXK381c!#@&kz,x,1.12Z3A12X2AZ1xX1",
+	"authentication_username" : "admin",
+	"authentication_password" : "admin",
 	"authentication_realm" : "KinashServer",
 	"authentication_file" : "./public_html/login.html",
 	"max_url_length": "500",
 	"max_url_length_tip_do_not_change": "Pro Tip: Use only numbers in max_url_length to prevent server crash",
```
Comment (andriycraft) [inline configs/config.json:10]: remove

**Review by andriycraft — CHANGES_REQUESTED** [review]

File: app.js (hunk 15 above)
Comment (andriy332) [reply app.js:128]: It must be returnError(500, null, null)

File: configs/config.json (hunk 23 above)
Comment (andriy332) [reply configs/config.json:10]: no

**Pull request merged**